### DISCUSSION TITLE
Reference regarding "unit testing rules" of outgoing queries

### DISCUSSION THREAD

**Discussion by robveijk** [pycabook/pycabook#50] [Q&A] [answered]
In [section 3.4](https://github.com/pycabook/pycabook/blame/master/src/05_Chapter_03_A_basic_example.adoc#L390) it is mentioned that "according to the unit testing rules, we should not test outgoing queries". I'm wondering why that is (is it because that would be considered an integration test)?
A reference in the text would really be helpful and improve the text (imho). Thanks!

Thanks for writing this awesome book 🙏  I'm slowly working through this and really enjoying it!

**Comment by lgiordani** [comment, marked as answer]
Hi @robveijk thanks for the message and for reading my book!

That was left unexplained because it was originally explained in a section that I removed from the book :see_no_evil: You can find an in-depth explanation here https://www.thedigitalcatonline.com/blog/2020/09/15/tdd-in-python-with-pytest-part-3/

Feel free to go on with the discussion if that post doesn't clarify the matter, though. I will add a clarification to the book, thanks!

**Reply by robveijk** [reply]
Hi @lgiordani 

Thanks for your quick and elaborate answer! I really should have read your pytest blog series first (especially as they were referred to as prior knowledge in your book). I definitely will now 😅 
Really clear explanation in https://www.thedigitalcatonline.com/blog/2020/09/15/tdd-in-python-with-pytest-part-3/#external-systems. Thanks! 😊 

All the best,
Rob
